crates/sui-core/src/authority.rs

```diff
@@ -409,6 +413,7 @@ pub struct AuthorityState {
 
     pub event_handler: Option<Arc<EventHandler>>,
     pub transaction_streamer: Option<Arc<TransactionStreamer>>,
+    checkpoint_store: Option<Arc<CheckpointStore>>,
```

Comment:
Why is this Option?

Reply:
I saw a `enable_checkpoint` config so I thought checkpoint_store should be None when `enable_checkpoint` is false, did I miss sth?

Reply:
We probably need to remove that config, as now our system doesn't work without checkpoints.

Reply:
That config is no longer used

Reply:
I see, I removed the option then. thanks for the contexts !